packages/plans-grid-next/src/hooks/use-upgrade-click-handler.ts

```diff
@@ -1,58 +0,0 @@
-import { isFreeHostingTrial, type PlanSlug } from '@automattic/calypso-products';
-import { WpcomPlansUI } from '@automattic/data-stores';
-import { useSelect } from '@wordpress/data';
-import { useCallback } from '@wordpress/element';
-import type { GridPlan } from '../types';
-import type { MinimalRequestCartProduct } from '@automattic/shopping-cart';
-
-interface Props {
-	gridPlans: GridPlan[]; // TODO clk: to be removed, grabbed from context
-	onUpgradeClick?: (
-		cartItems?: MinimalRequestCartProduct[] | null,
-		clickedPlanSlug?: PlanSlug
-	) => void;
-}
-
-const useUpgradeClickHandler = ( { gridPlans, onUpgradeClick }: Props ) => {
-	const selectedStorageOptions = useSelect( ( select ) => {
-		return select( WpcomPlansUI.store ).getSelectedStorageOptions();
-	}, [] );
-
-	return useCallback(
-		( planSlug: PlanSlug ) => {
-			const selectedStorageOption = selectedStorageOptions?.[ planSlug ];
-			const { cartItemForPlan, storageAddOnsForPlan } =
-				gridPlans.find( ( gridPlan ) => gridPlan.planSlug === planSlug ) ?? {};
-			const storageAddOn = storageAddOnsForPlan?.find( ( addOn ) => {
-				return selectedStorageOption && addOn
-					? addOn.featureSlugs?.includes( selectedStorageOption )
-					: false;
-			} );
-			const storageAddOnCartItem = storageAddOn &&
-				! storageAddOn.purchased && {
-					product_slug: storageAddOn.productSlug,
-					quantity: storageAddOn.quantity,
-					volume: 1,
-					extra: { feature_slug: selectedStorageOption },
-				};
```

Comment:
We should probably have a slightly better interface for the plan-action callbacks, one that would also accept a set of options - and one of them being the "selected add-on from the grid", similar to what's done in this hook. 

We always have to think about "enabling" functionality for logged-out pages with these refactors. The grid can do a little more to make its interface simpler. Add-ons are part of the grid, they are being selected, and any call to action would obviously need to account for that selection. The add-ons will be coming to `/pricing` soon too (and that's the ultimate goal with the refactors, just straight up import and use in logged-out - not go through a ton of setting up).

Yes, we are populating a store with the selection, but do we want to have to create logic to deal with that side-effect? More so if the callbacks are plain functions, we'd probably need additional refactors too.

For now, I think let's add/keep a slim callback handler hook in `plans-grid-next` to query the store and pass along the selected add-on into these callbacks.